use-case-guides/workday-request-time-off/README.md

```diff
@@ -209,149 +186,346 @@ curl --location 'https://{{domain}}.workday.com/ccx/api/absenceManagement/v1/{{i
+    Returns:
+        JSON response from the time off request API.
+    """
+    try:
+        response = await request_time_off(
+            email=email,
+            start_date=time_off_request.start_date,
+            time_off_type_id=time_off_request.time_off_type_id,
+            end_date=time_off_request.end_date,
+            comment=time_off_request.comment,
+        )
+        return response
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+@app.get("/")
+async def root():
+    """
+    Root endpoint to test the API.
     
-    response = requests.post(url_request_off, json=payload, headers=headers)
-    if response.status_code == 200:
-        return {"message": "Time off requested successfully."}
-    else:
-        return {"error": f"Request failed with status code {response.status_code}: {response.text}"}
+    Returns:
+        JSON response which validates the API is working.
+    """
+    content = {"text": "Hello World"}
+    headers = {"Content-Type": "application/json"}
+    return JSONResponse(content=content, headers=headers)
+
 ```
 
+## Step 4: Build in Creator Studio
+
+We will be building this plugin as a Path along with an iPaaS to chain APIs 1, 2, and 3 together in order to submit a time off request for a user.
+
+### Plugin Architecture Diagram
+
+![Untitled](./Plugin%20Template%20Request%20Time-Off%20in%20Workday%20081c4d522bf64bbead3697288dd46047/Untitled.png)
+
+**Triggers**
+
+1. Natural language input to copilot requesting to take time off
+
+**Natural Language Slots**
+
+1. Start Date: when time off starts
+2. End Date *(optional)*: when time off ends. By default end date is the same as the start date.
+3. Comments *(optional)*: Any relevant context / justification for taking time off
+
+**Actions**
+
+1. Lookup time off type details: Executes the `/api/wql/v1/{{instance}}/data` API to query the requesting employee's Workday ID, eligible time off plans, and relevant time off types through their work email address.
+2. Request time off: Executes the `/api/absenceManagement/v1/{{instance}}/workers/{{worker_id}}/requestTimeOff` API to request time off
+
+**Caveats**
+
+1. If an employee selects a time off plan without adequante time balance, you will get an error message for attempting to request time off from the REST API. You can personalize the error message as part of exception handling logic inside your iPaaS.
+2. An employee could specify a time off plan that they are not eligible for
+1. Current Date: used in the "as of date" in the request PTO action. Could be static.
+2. Email Address: we assume that a user is only requesting time off for themselves, and use the `{{user.email_addr}}` value from user roster profile of the employee calling this plugin. This would need to be changed if an employee is requesting time off on behalf of another employee.
+
+### Build Plugin in Creator Studio
+1. Create a Path in Creator Studio. You can copy the plugin details below
```

Comment:
Wait a sec, your earlier architecture diagram said `requires resolver`, but here you're doing a path based use case....

Reply:
And then your ipaas code expects a time off type ID

Reply:
These two approaches are inconsistent in the guide

Reply:
you don't need to maintain a mapping of time off type -> time off type id in the ipaas – if you're building a path you can simply pass the id directly – see implementation https://m8.moveworks.com/creator-studio/paths/flow.edit?uuid=3ed01a4b-e77c-43ef-9f2d-8b0b9bf1d1ff